code/game/machinery/syndicatebeacon.dm

```diff
@@ -80,14 +83,21 @@
 
 //stealth direct power usage
 /obj/machinery/power/singularity_beacon/process()
+	say("Process")
 	if(!active)
 		return PROCESS_KILL
 	else
 		if(surplus() > 1500)
 			add_load(1500)
+			if(cooldown <= world.time)
+				cooldown = world.time + 100
+				for(var/obj/singularity/singulo in world)

```

Comment:
Use poi_list instead of world, solves the problem and makes this faster. Singularities could probably use their own list.